contracts/TokenManagerHook.sol

```diff
@@ -4,21 +4,38 @@
 
 pragma solidity 0.4.24;
 
+import "@aragon/os/contracts/common/ReentrancyGuard.sol";
 
-interface ITokenManagerHook {
+
+/**
+* @dev When creating a subcontract, we recommend overriding the _internal_ functions that you want to hook.
+*/
+contract TokenManagerHook is ReentrancyGuard {
+    address private tokenManager;
+
+    modifier onlyTokenManager () {
+        require (tokenManager == msg.sender, "Hooks must be called from Token Manager");
+        _;
+    }
     /*
     * @dev Called when this contract has been included as a Token Manager hook
     * @param _tokenManager Token Manager instance that has included the hook
     * @param _hookId The position in which the hook is going to be called
     */
-    function onRegisterAsHook(address _tokenManager, uint256 _hookId) external;
+    function onRegisterAsHook(uint256 _hookId) external nonReentrant {
```

Comment:
@willjgriff Do you think that applying the `nonReentrant` modifier here is effective? It basically prevents calling a function that triggers a hook from within a hook. I don't know if it is a good solution to prevent possible inconsistencies in the future, or I'm just limiting what people can do with hooks. What is your opinion?

Reply:
It's a good point, maybe a `transfer()` needs to do another `transfer()`. However, if a contract needs to do this they can just implement their own TokenManagerHook without the `nonReentrant` modifier. So I think this is good.